internal/app/storage/storagejson.go

```diff
@@ -0,0 +1,87 @@
+package storage
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+)
+
+// структура хранилища
+type StorageJs struct {
+	IDURL    map[string]string `json:"idurl,omitempty"`
+	pathName string
+}
+
+// метод записи id:url в хранилище
+func (ms *StorageJs) PutStorage(key string, value string) (err error) {
+	if value, ok := ms.IDURL[key]; ok {
+		return fmt.Errorf("key %s is already in database", value)
+	}
+	ms.IDURL[key] = string(value)
+	// открываем файл
+	sfile, err := os.OpenFile(ms.pathName, os.O_WRONLY, 0777)
+	if err != nil {
+		log.Println("storage file opening error: ", err)
+		return err
+	}
+	defer sfile.Close()
+	// кодирование в JSON
+	js, err := json.Marshal(&ms.IDURL)
+	if err != nil {
+		log.Println("JSON marshalling from struct error: ", err)
+		return err
+	}
+	// запись в файл
+	sfile.Write(js)
+	return nil
+}
+
+// конструктор нового хранилища JSON
+func NewJsStorage(s map[string]string, p string) *StorageJs {
+	// загрузка базы из JSON
+	_, pathOk := os.Stat(filepath.Dir(p))
+
+	if os.IsNotExist(pathOk) {
+		os.MkdirAll(filepath.Dir(p), 0777)
+		log.Printf("folder %s created\n", filepath.Dir(p))
+	}
+	sfile, err := os.OpenFile(p, os.O_RDONLY|os.O_CREATE, 0777)
+	if err != nil {
+		log.Fatal("file creating error: ", err)
+	}
+	defer sfile.Close()
+
+	fileInfo, _ := os.Stat(p)
+	if fileInfo.Size() != 0 {
+		b, err := io.ReadAll(sfile)
+		if err != nil && err != io.EOF {
```

Comment:
Вроде бы для ReadAll EOF не надо проверять:
```
// ReadAll reads from r until an error or EOF and returns the data it read.
// A successful call returns err == nil, not err == EOF. Because ReadAll is
// defined to read from src until EOF, it does not treat an EOF from Read
// as an error to be reported.
```

Reply:
да, осталось от другого подхода)